fwup-ops.conf

```diff
@@ -74,6 +90,63 @@ uboot-environment uboot-env {
     block-count = ${UBOOT_ENV_COUNT}
 }
 
+##
+# factory-reset
+##
+task factory-reset {
+    on-init {
+        info("Erasing all writable data")
+        # This requires --enable-trim
+        # Trim may not work on MicroSD card, so don't rely on it
+        trim(${APP_PART_OFFSET}, ${APP_PART_COUNT})
+        raw_memset(${APP_PART_OFFSET}, 256, 0xff)
+    }
+}
+
+##
+# prevent-revert
+#
+# Pass `--enable-trim` to also clear out the partition that no longer should be used.
+##
+task prevent-revert.a {
+    # Check that we're running on B
+    require-partition-offset(0, ${BOOT_B_PART_OFFSET})
+    require-partition-offset(1, ${ROOTFS_B_PART_OFFSET})
+    require-uboot-variable(uboot-env, "nerves_fw_active", "b")
+
+    on-init {
+        info("Preventing reverts to partition A")
+        # Remove U-Boot variables that fwup uses to allow reverting images
+        uboot_unsetenv(uboot-env, "a.nerves_fw_platform")
+        uboot_unsetenv(uboot-env, "a.nerves_fw_architecture")
+        # Clear out the old image using TRIM. This requires --enable-trim
+        trim(${ROOTFS_A_PART_OFFSET}, ${ROOTFS_A_PART_COUNT})
```

Comment:
I don't know if more is required here. I'm still trying to figure out how (if) the MQ pro has 2 different boot paths (like how bbb has `zImage.a` and `zImage.b` that can be cleared)